File: backend/src/NaviFreight.Api/Models/AuthModels.cs
namespace NaviFreight.Api.Models;

public sealed record LoginRequest(string Email, string Password);

public sealed record LoginResponse(
    string Token,
    DateTime ExpiresAt,
    UserProfile User);

public sealed record UserProfile(
    int UserId,
    string Email,
    string DisplayName,
    string Role,
    string TenantId);

public sealed record UserIdentity(
    int UserId,
    string TenantId,
    string Email,
    string DisplayName,
    string Role,
    string PasswordHash);

File: frontend/src/app/core/models/auth.models.ts
export type UserRole = 'Tenant Admin' | 'Dispatcher' | 'Yard Manager';

export interface LoginRequest {
  email: string;
  password: string;
}

export interface UserProfile {
  userId: number;
  email: string;
  displayName: string;
  role: UserRole;
  tenantId: string;
}

export interface LoginResponse {
  token: string;
  expiresAt: string;
  user: UserProfile;
}
